File: ai-ml/lesson-05-data-visualization/hard/p01-solve.py
"""
LESSON — Data Visualization
HARD P01 — Full EDA function
==================================================

CONCEPT:
  See concepts.md in this lesson folder for detailed explanations.

PROBLEM:
  Write a function that takes a DataFrame and performs complete EDA: prints shape, dtypes, missing values, and describe(); creates and saves a figure with 4 subplots (histogram of a numeric column, box plot by a categorical column, scatter of two numeric columns, and a correlation heatmap). Return a summary dictionary


Write a function `solve()` that implements the solution.

# Check your answer: compare with solutions/p01-solution.py
"""

TRY THIS INPUT:
  Add this test code at the bottom of your file:

  ```python
  n = 300
  df = pd.DataFrame({
  df.loc[df.sample(20).index, "income"] = np.nan
  df.loc[df.sample(10).index, "score"] = np.nan

  # Call your function
  result = solve(...)
  print(result)
  ```

EXPECTED OUTPUT:
  Run the solution file to see expected output:
    python3 hard/solutions/p01-solution.py

  Then compare your output format with theirs.

# === WRITE YOUR CODE BELOW ===
# TODO: Write your solution from scratch.

File: html-css-javascript/lesson-19-js-objects-dom/medium/p01-solve.js
/**
LESSON 19 — JavaScript Objects & DOM
=====================================

PROBLEM: Merge Configs (Medium)
Write mergeConfigs(defaults, userPrefs) with spread — prefs win.

TRY THIS:
  - return { ...defaults, ...userPrefs }.
  - Test with defaults {theme, fontSize, lang} and prefs overriding
    two of them.

EXPECTED OUTPUT:
  mergeConfigs({theme:"light",fontSize:14,lang:"en"},
               {theme:"dark",fontSize:16})
    -> { theme: "dark", fontSize: 16, lang: "en" }

TEST: node medium/p01-solve.js

CHECK: python3 check.py medium/p01
*/

// TODO: Write your complete solution from scratch below.
